### PR TITLE
Remove TextFile from public API

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes [this comment](https://www.reddit.com/r/godot/comments/f2pkyj/godot_proposal_use_new_financial_resources_to/fheos9v/)
(XD)

We should probably think of some way to hide internal classes like that, because they are useless for scripting anyways.

### CODE REVIEW

**Comment by akien-mga** [comment]
> We should probably think of some way to hide internal classes like that, because they are useless for scripting anyways.

Technically, I think you can `load()` a file type declared as handled as `TextFile` and that will be its type, so some code like `load("res://myfile.json") is TextFile` might be valid.

Not that there's be much use case for it, but that's why it makes sense for it to be exposed, since it is part of the public API.

File: doc/classes/TextFile.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="TextFile" inherits="Resource" version="4.0">
 	<brief_description>
+		A resource representation of plain text file.
```
Comment (clayjohn) [inline doc/classes/TextFile.xml]: "of _a_ plan text file"

File: doc/classes/TextFile.xml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 	</brief_description>
 	<description>
+		[TextFile] is used internally by script editor when a plain text file is edited. Normally it's not possible to load text files from code as a resource, so this class is not intended for script usage. If you want to load a file as text, use [method File.get_as_text].
```
Comment (clayjohn) [inline doc/classes/TextFile.xml]: "is used internally be _the_ script editor"

**Comment by KoBeWi** [comment]
The thing is that you can't do `load("res://myfile.json")`, because it gives "No loader found" error. The class is basically useless, because you can't even create a valid instance. It doesn't expose any way to access the text either.

**Review by Paulb23 — COMMENTED** [review]
Yeah `TextFile` should never be used outside the editor. It's not even a "proper" resource as everything related to is handled directly by the `ScriptEditorPlugin`, goes nowhere near the normal route of `ResourceLoader/Saver`.

**Comment by akien-mga** [comment]
Then removing it's `GDCLASS` call might do the trick.

**Comment by akien-mga** [comment]
> Remove TextFile 

... "from public API" would be good. You're not removing `text_file.h` :)

**Comment by KoBeWi** [comment]
I removed `GDCLASS` and it seems to be ok.

**Comment by akien-mga** [comment]
Yeah squashed it should be good to merge.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Paulb23** [comment]
This breaks `TextFiles` as they can no longer be opened, not sure how else to mark it as editor only :/

**Comment by akien-mga** [comment]
Heh, I assumed @KoBeWi would have tried that... :D

**Comment by KoBeWi** [comment]
I tried and the files were still opening.

**Comment by Paulb23** [comment]
I see a lot of `ERROR: Cannot get class 'TextFile'.` printed to the output.

**Comment by KoBeWi** [comment]
Ah, I didn't look in there. Well, the files do open, so maybe it's possible to remove the error without exposing the class again.

**Pull request merged**